lib/std/static_string_map.zig

```diff
@@ -1,446 +1,170 @@
-            if (self.kvs.len == 0)
-                return null;
-
-            if (str.len < self.min_len)
-                return null;
-
-            var len = @min(self.max_len, str.len);
-            while (len >= self.min_len) : (len -= 1) {
-                if (self.getIndex(str[0..len])) |i|
-                    return i;
+        /// Filters the input key by length, then compares it to the possible matches.
+        /// Because we know the length at comptime, we can compare the strings faster.
+        fn filterLength(comptime self: Self, key: []const T) ?V {
+            // Provide 2000 branches per key/value pair to compile.
+            @setEvalBranchQuota(2000 * self.kvs.len);
+            const kvs_by_lengths = comptime self.separateLength();
+            inline for (kvs_by_lengths) |kvs_by_len| {
+                const len = kvs_by_len.length;
+                if (key.len == len) {
+                    inline for (kvs_by_len.kvs) |kv| {
+                        if (eql(T, len, kv.key[0..len].*, key[0..len].*)) {
+                            return kv.value;
+                        }
+                    }
+                }
             }
             return null;
         }
 
-        pub fn keys(self: Self) []const []const u8 {
-            const kvs = self.kvs.*;
-            return kvs.keys[0..kvs.len];
-        }
+        const LengthKvs = struct {
+            length: usize,
+            kvs: []const Kv,
+        };
+
+        /// Creates a list of kv sets grouped by different key lengths.
+        fn separateLength(comptime self: Self) []const LengthKvs {
+            var length_sets: []const LengthKvs = &.{};
+            add_length: for (self.kvs, 0..) |check_kv, index| {
+                const length = check_kv.key.len;
+
+                // Skip the current length if it has already been inserted.
+                for (length_sets) |set| {
+                    if (set.length == length) {
+                        continue :add_length;
+                    }
+                }
 
-        pub fn values(self: Self) []const V {
-            const kvs = self.kvs.*;
-            return kvs.values[0..kvs.len];
+                var added_kvs: []const Kv = &.{};
+                for (self.kvs[index..]) |add_kv| {
+                    if (add_kv.key.len == length) {
+                        added_kvs = added_kvs ++ .{add_kv};
+                    }
+                }
+                length_sets = length_sets ++ .{.{ .length = length, .kvs = added_kvs }};
+            }
+            return length_sets;
         }
     };
 }
 
-const TestEnum = enum { A, B, C, D, E };
-const TestMap = StaticStringMap(TestEnum);
-const TestKV = struct { []const u8, TestEnum };
-const TestMapVoid = StaticStringMap(void);
-const TestKVVoid = struct { []const u8 };
-const TestMapWithEql = StaticStringMapWithEql(TestEnum, eqlAsciiIgnoreCase);
 const testing = std.testing;
-const test_alloc = testing.allocator;
-
-test "list literal of list literals" {
```

Comment:
I would suggest trying to restore all of these tests where possible.

Reply:
Good suggestion for the tests - I had a latent logic error :)
I was planning to move most of them back, but had forgotten.